packages/jest-haste-map/src/index.ts

```diff
@@ -219,7 +229,22 @@ export default class HasteMap extends EventEmitter {
   private _watchers: Array<Watcher>;
   private _worker: WorkerInterface | null;
 
-  constructor(options: Options) {
+  static getStatic(config: Config.ProjectConfig): HasteMapStatic {
```

Comment:
can we make this `async` so we can use `import(config.haste.hasteMapModulePath)` in the future? same for `create`

Reply:
right, but it needs to be async in order to use ESM. so we either need to make it so now, or make it async whenever somebody wants to write these modules using ESM (such as a WASM implementation)